### PR TITLE
DAppNode package

### PR DESCRIPTION
**Pull request by TripleSpeeder**
**UPDATED PR**
There is now a separate compose file for building the DAppNode package, instead of trying to use the same compose file.

The changes now should be fully compatible to the existing setup, we can run & build local images or images for the docker.com registry the same way like before. 

PR is now ready for review :-)

## Overview of changes:
- Make arango db connection configurable through ENV for all packages (BN_ARANGO_PROTOCOL, BN_ARANGO_HOST, BN_ARANGO_PORT)
- consensus packages: The `entrypoint` parameter in docker_compose.yml does not work with dappnodeSDK, so i created a new entrypoint.sh and ENV var "RUN_TYPE" that is used to decide if consensus should run in `sender` or `receiver` mode
- db package:
  - make listen address configurable in entrypoint script through ENV var BN_ARANGO_SERVER_ENDPOINT. 
  - entrypoint script changed so it detects if this is an initial install (requiring download of brightID dump) or update install, without having to set INIT_BRIGHTID_DB variable. You can still set INIT_BRIGHTID_DB = 1 to force it.
- Add necessary files to build DAppNode package:
  - docker-compose_DAppNode.yml
  - README_dappnode.md
  - dappnode_package.json
  - default avatar

### CODE REVIEW

**Review by abramsymons — COMMENTED** [review]
Some updates applied including:
- Using `ENGINE` file created by arangodb in the first run instead of custom `db_initialized` to prevent re-initialization of nodes (especially node one) when upgrading to this release
- Enable operators running read-only nodes by not setting `BN_CONSENSUS_PRIVATE_KEY`
- Move `brightid-nginx.conf` to `web` folder just like `brightid-nginx-dappnode.conf`
- Use `python-arango==5.4.0` for all python services to enable passing `hosts` to `ArangoClient`

File: db/docker-entrypoint.sh [hunk 1]
```diff
@@ -2,7 +2,7 @@
 # Edited from https://github.com/arangodb/arangodb-docker/blob/official/alpine/3.6.4/docker-entrypoint.sh
 set -e
 
-if [ "$INIT_BRIGHTID_DB" == "1" ] || [ ! -f /var/lib/arangodb3/db_initialized ]; then
+if [ "$INIT_BRIGHTID_DB" == "1" ] || [ ! -f /var/lib/arangodb3/ENGINE ]; then
```
Comment (adamstallard) [inline db/docker-entrypoint.sh:5]: What's the `ENGINE` file?

File: consensus/docker-entrypoint.sh [hunk 2]
```diff
@@ -4,8 +4,12 @@ set -e
...
 
 if [ "$RUN_TYPE" = "SENDER" ] ; then
-  echo "Starting in SENDER mode"
-  exec python3 -u sender.py
+  if [ "$BN_CONSENSUS_PRIVATE_KEY" != "" ] ; then
+    echo "Starting in SENDER mode"
+    exec python3 -u sender.py
+  else
+    echo "Ignore starting sender because BN_CONSENSUS_PRIVATE_KEY is not set"
+  fi
```
Comment (adamstallard) [inline consensus/docker-entrypoint.sh:12]: Why does it matter whether or not a read-only node starts sender?  Are we try to saving resources from not running this service or are we trying to save operator time by not requiring generating a key pair they're not going to use?

File: db/docker-entrypoint.sh (hunk 1 above)
Comment (abramsymons) [reply db/docker-entrypoint.sh:5]: It's one of the files that arangod saves on `data` named volume when the volume is created and db arangod runs for the  first time.

```
$ docker exec -it brightid-node_db_1 /bin/sh
/ # ls /var/lib/arangodb3
ENGINE          LANGUAGE        LOCK            SERVER          VERSION-1       engine-rocksdb
/ # 
```
I understand that there is a risk of having this file missed when we upgrade our arangodb version that can result in re-initialization of nodes but I did not find a better way to find out if `db` is initialized before or not. If we want to use this approach, we should be careful when upgrading our arango version.

File: consensus/docker-entrypoint.sh (hunk 2 above)
Comment (abramsymons) [reply consensus/docker-entrypoint.sh:12]: In fact @TripleSpeeder introduced the `read-only node` term in Dappnode package README file.
I think the main reason was to save operator time by not requiring generating a key pair they're not going to use and sending Eidi to that.
I just wanted added this to ensure exceptions will not be raised from `consensus_sender` if `BN_CONSENSUS_PRIVATE_KEY` var is not which is current dapp node configuration.

**Pull request merged**